netbench/cdk/src/main/java/com/aws/ClientServerStack.java

```diff
@@ -0,0 +1,115 @@
+import software.amazon.awscdk.services.lambda.Function;
+import software.amazon.awscdk.services.lambda.Runtime;
+import software.amazon.awscdk.RemovalPolicy;
+
+import java.util.HashMap;
+
+
+public class ClientServerStack extends Stack {
+    private String cidr;
+    private Vpc vpc;
+
+    public ClientServerStack(final Construct parent, final String id) {
+        this(parent, id, null);
+    }
+
+    public ClientServerStack(final Construct parent, final String id, final ClientServerStackProps props) {
+        super(parent, id, props);
+
+        String instanceType = props.getInstanceType();
+        String stackType  = props.getStackType();
+        this.cidr = props.getCidr();
+
+        this.vpc = Vpc.Builder.create(this, stackType + "-vpc")
+            .maxAzs(1)
+            .cidr(cidr)
+            .build();
+
+        SecurityGroup.fromSecurityGroupId(this, "vpc-sec-group", this.vpc.getVpcDefaultSecurityGroup())
+            .addIngressRule(Peer.anyIpv4(), Port.icmpPing(), "Allow ping anywhere.");
+
+        GatewayVpcEndpoint s3Endpoint = this.vpc.addGatewayEndpoint("s3-endpoint",
+            GatewayVpcEndpointOptions.builder()
+            .service(GatewayVpcEndpointAwsService.S3)
+            .build());
+
+        StringParameter.Builder.create(this, stackType + "-vpc-id")
+            .parameterName(stackType + "-vpc-id")
+            .stringValue(this.vpc.getVpcId())
+            .build();
+
+        StringParameter.Builder.create(this, stackType + "-cidr")
+            .parameterName(stackType + "-cidr")
+            .stringValue(this.vpc.getVpcCidrBlock())
+            .build();
+
+        /*
+        Cluster cluster = Cluster.Builder.create(this, stackType + "-cluster")
+            .vpc(vpc)
+            .build();
+        
+        AutoScalingGroup asg = AutoScalingGroup.Builder.create(this, stackType + "-asg")
+            .vpc(vpc)
+            .instanceType(new InstanceType(instanceType))
+            .machineImage(EcsOptimizedImage.amazonLinux2())
+            .minCapacity(0)
+            .build();
+
+        AsgCapacityProvider asgProvider = AsgCapacityProvider.Builder.create(this, stackType + "-asg-provider")
+            .autoScalingGroup(asg)
+            .build();
+        
+        cluster.addAsgCapacityProvider(asgProvider);
+
+        Ec2TaskDefinition task = Ec2TaskDefinition.Builder
+            .create(this, stackType + "-task")
+            .build();
+        task.addContainer(); 
+        
+        Ec2Service.Builder.create(this, "ec2service-" + stackType)
+            .cluster(cluster)
+            .taskDefinition(task)
+            .build();
+        */
+
+        BastionHostLinux testInstance = BastionHostLinux.Builder.create(this, "testInstance")
+            .vpc(vpc)
+            .securityGroup(SecurityGroup.fromSecurityGroupId(this, stackType + "vpc-sec-group", this.vpc.getVpcDefaultSecurityGroup()))
+            .build();
+
+        Bucket metricsBucket = props.getBucket();
```

Comment:
These don't seem to be used?

Reply:
Yeah, the ec2 instance is a placeholder for the ECS tasks that I'm working on right now to show that my VPC connections work. The metrics bucket will be used for report generation later on.